hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/management/GetCPMembersMessageTask.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task.management;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.MCGetCPMembersCodec;
+import com.hazelcast.client.impl.protocol.codec.MCGetCPMembersCodec.RequestParameters;
+import com.hazelcast.client.impl.protocol.task.AbstractAsyncMessageTask;
+import com.hazelcast.cp.CPMember;
+import com.hazelcast.cp.CPSubsystemManagementService;
+import com.hazelcast.instance.impl.Node;
+import com.hazelcast.internal.management.ManagementCenterService;
+import com.hazelcast.internal.nio.Connection;
+
+import java.security.Permission;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+public class GetCPMembersMessageTask extends AbstractAsyncMessageTask<RequestParameters, List<SimpleEntry<String, String>>> {
+
+    public GetCPMembersMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
+        super(clientMessage, node, connection);
+    }
+
+    @Override
+    protected CompletableFuture<List<SimpleEntry<String, String>>> processInternal() {
+        CPSubsystemManagementService cpService =
+                nodeEngine.getHazelcastInstance().getCPSubsystem().getCPSubsystemManagementService();
+        return cpService.getCPMembers().toCompletableFuture()
+                .thenApply(cpMembers -> {
+                    List<SimpleEntry<String, String>> result = new ArrayList<>();
```

Comment:
Why do we have `List<Entry<String, String>>` here instead of `CPMemberDTO`?

Reply:
That's because we have to deal with data types defined for codecs. And the general guideline in the client protocol generator is to avoid adding custom types where possible. That's why `List<CPMember>` is represented as `List<Map.Entry<String, String>>`.

A side note: it's going to change from `List<Map.Entry<String, String>>` to `List<Map.Entry<UUID, Address>>` (see comment from @sancar).

Reply:
got it. thx